### ISSUE TITLE
heasarc table parsing needs format

### ISSUE THREAD

**Issue by caseyjlaw** [astropy/astroquery#1575]
There seems to have been a change in astropy and/or astroquery that broke heasarc queries.

My setup:
```
> conda list | grep astro
astropy                   3.2.2                    pypi_0    pypi
astroquery                0.3.10                   pypi_0    pypi
```

My error:
```
In [1]: from astroquery import heasarc                                                                                                                                                      
In [2]: from astropy import coordinates, units                                                                                                                                              
In [3]: co = coordinates.SkyCoord('1h0m0s 0d0m0s', unit=(units.hour, units.deg))                                                                                                            
In [4]: he = heasarc.Heasarc()                                                                                                                                                              
In [5]: he.query_region(co, mission='NVSS', radius =1*units.deg)                                                                                                                            
---------------------------------------------------------------------------
IORegistryError                           Traceback (most recent call last)
<ipython-input-5-88ba85f13f18> in <module>
----> 1 he.query_region(co, mission='NVSS', radius =1*units.deg)

~/anaconda/envs/rfs/lib/python3.6/site-packages/astroquery/utils/class_or_instance.py in f(*args, **kwds)
     23         def f(*args, **kwds):
     24             if obj is not None:
---> 25                 return self.fn(obj, *args, **kwds)
     26             else:
     27                 return self.fn(cls, *args, **kwds)

~/anaconda/envs/rfs/lib/python3.6/site-packages/astroquery/utils/process_asyncs.py in newmethod(self, *args, **kwargs)
     27             if kwargs.get('get_query_payload') or kwargs.get('field_help'):
     28                 return response
---> 29             result = self._parse_result(response, verbose=verbose)
     30             self.table = result
     31             return result

~/anaconda/envs/rfs/lib/python3.6/site-packages/astroquery/heasarc/core.py in _parse_result(self, response, verbose)
    220         try:
    221             data = BytesIO(response.content)
--> 222             table = Table.read(data, hdu=1)
    223             return table
    224         except ValueError:

~/anaconda/envs/rfs/lib/python3.6/site-packages/astropy/table/connect.py in __call__(self, *args, **kwargs)
     50     def __call__(self, *args, **kwargs):
     51         cls = self._cls
---> 52         out = registry.read(cls, *args, **kwargs)
     53 
     54         # For some readers (e.g., ascii.ecsv), the returned `out` class is not

~/anaconda/envs/rfs/lib/python3.6/site-packages/astropy/io/registry.py in read(cls, format, *args, **kwargs)
    518 
    519             format = _get_valid_format(
--> 520                 'read', cls, path, fileobj, args, kwargs)
    521 
    522         reader = get_reader(format, cls)

~/anaconda/envs/rfs/lib/python3.6/site-packages/astropy/io/registry.py in _get_valid_format(mode, cls, path, fileobj, args, kwargs)
```

**Comment by caseyjlaw** [comment]
Just an update: none of the heasarc queries in the docs work. Even `heasarc.query_mission_list()` returns a table parsing error. Perhaps there was a change on the heasarc server side?

**Comment by bsipocz** [comment]
> Perhaps there was a change on the heasarc server side?

yes, most probably, tests are failing, too (they worked at the time of v0.3.10 release)

**Comment by bsipocz** [comment]
OK, so the above query returns a 404 not found, rather than a fits file, definitely some changes upstream.

@keflavich - do we have anyone to ping, or it's a fend for ourselves situation?

**Comment by keflavich** [comment]
We might be on our own, but let's see if we can identify someone to help...

**Comment by keflavich** [comment]
Is it this? https://heasarc.gsfc.nasa.gov/docs/FTPWarning.html

**Comment by bsipocz** [comment]
Not exactly, but the gist of it is the same.

**Comment by caseyjlaw** [comment]
I see that the astroquery heasarc module defines a url ('conf') as `http://heasarc.gsfc.nasa.gov/cgi-bin/W3Browse/w3query_noredir.pl`. That page no longer exists and a message suggests contacting Stephan.Fantasia@nasa.gov.
Shall I reach out to this person?

**Comment by ellot** [comment]
I change the server as "https://heasarc.gsfc.nasa.gov/db-perl/W3Browse/w3query.pl". Then it works. 

Actually, in the __init__.py file of heasarc module, it has mentioned that the old server does not work and defined the "mission_server". Modifing the "mission_server" to "server" can resolve this issue.

**Comment by bsipocz** [comment]
@ellot - since you did this investigation, would you be interested in opening a pull request to update the urls?

**Comment by keflavich** [comment]
Closed by #1576.  Thanks @ellot and @matteobachetti
